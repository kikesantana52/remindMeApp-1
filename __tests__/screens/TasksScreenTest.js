import 'react-native';
import React from 'react';
import TasksScreen from '../../src/screens/TasksScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
// instanciar para llamar funciones/variables y estados
const Screen = new TasksScreen();
let component;
const navigationMock = { navigate: jest.fn() };

describe('Tasks Screen - ', () => {
  // Que si entre al screen y reenderie bien
  it('should render correctly', () => {
    component = renderer.create(
      <TasksScreen navigation={navigationMock}/>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should change state to true',()=>{
    const obj = new TasksScreen();
    obj.openAddTaskModal();
    let resp = [...obj.state.showAddTaskModal];
    expect(resp).toBeTruthy();

  });

  it('should change state to false',()=>{
    const obj = new TasksScreen();
    obj.closeAddTaskModal();
    let resp = obj.state.showAddTaskModal;
    expect(resp).toBeFalsy();

  });

  it('should say how much tasks are need to be completed',()=>{
    let tasks= [...Screen.state.tasks];
    expect(Screen.calcultateToBeCompletedTasks(tasks)).toEqual(3);
  });

  it('should toogle the flag of completed of the task that I use',()=>{
    Screen.toggleTask(1);
    let tasks= [...Screen.state.tasks];
    expect(Screen.calcultateToBeCompletedTasks(tasks)).toEqual(2);
  });

  it('should add a task',()=>{
    const obj = new TasksScreen();
    obj.addTask("Pasear el perro");
    console.log(obj.state.tasks);
    expect(Object.keys(obj.state.tasks).length).toEqual(4);
  });


});
