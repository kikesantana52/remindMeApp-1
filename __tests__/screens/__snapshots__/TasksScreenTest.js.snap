// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks Screen -  should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(255,255,255)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 399,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 399,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 399,
              "width": 750,
            },
            undefined,
          ]
        }
      />
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "borderRadius": 65,
            "height": 130,
            "width": 130,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgb(255,255,255)",
            "fontFamily": "Avenir",
            "fontSize": 36,
          }
        }
      >
        3
         Pendientes
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgb(164, 164, 164)",
            "fontFamily": "Avenir",
            "fontSize": 16,
          }
        }
      >
        VIERNES 15 DE MARZO DE 2018
      </Text>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "width": 750,
      }
    }
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 60,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "borderBottomColor": "rgb(231,231,231)",
              "borderBottomWidth": 2,
              "flex": 1,
              "flexDirection": "row",
              "height": 60,
              "padding": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "height": 20,
                "marginRight": 10,
                "marginTop": 4,
                "width": 20,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgb(29,29,38)",
                "fontFamily": "Avenir",
                "fontSize": 21,
              }
            }
          >
            Desayuno de trabajo
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 60,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "borderBottomColor": "rgb(231,231,231)",
              "borderBottomWidth": 2,
              "flex": 1,
              "flexDirection": "row",
              "height": 60,
              "padding": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "height": 20,
                "marginRight": 10,
                "marginTop": 4,
                "width": 20,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgb(29,29,38)",
                "fontFamily": "Avenir",
                "fontSize": 21,
              }
            }
          >
            Desarrollar App
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 60,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "borderBottomColor": "rgb(231,231,231)",
              "borderBottomWidth": 2,
              "flex": 1,
              "flexDirection": "row",
              "height": 60,
              "padding": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "height": 20,
                "marginRight": 10,
                "marginTop": 4,
                "width": 20,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgb(29,29,38)",
                "fontFamily": "Avenir",
                "fontSize": 21,
              }
            }
          >
            Dise√±ar Mocks
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(255,51,102)",
        "borderRadius": 72,
        "bottom": 20,
        "flex": 1,
        "height": 66,
        "justifyContent": "center",
        "position": "absolute",
        "right": 20,
        "width": 66,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 28,
          "width": 28,
        }
      }
    />
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,.5)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "height": 567,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(237,24,74)",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Image
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "width": 285,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": 315,
              "justifyContent": "space-around",
              "padding": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "Avenir",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            Agregar Tarea
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Nombre de la tarea"
            style={
              Object {
                "borderBottomColor": "rgb(231,231,231)",
                "borderBottomWidth": 2,
                "fontFamily": "Avenir",
                "fontSize": 30,
                "textAlign": "center",
                "width": 300,
              }
            }
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "rgb(237,24,74)",
                "borderColor": "rgb(237,24,74)",
                "borderRadius": 27,
                "borderStyle": "solid",
                "borderWidth": 2,
                "height": 55,
                "paddingTop": 15,
                "width": 325,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Avenir",
                  "fontSize": 17,
                  "textAlign": "center",
                }
              }
            >
              AGREGAR
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "rgb(237,24,74)",
                "borderRadius": 27,
                "borderStyle": "solid",
                "borderWidth": 2,
                "height": 55,
                "paddingTop": 15,
                "width": 325,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "rgb(237,24,74)",
                  "fontFamily": "Avenir",
                  "fontSize": 17,
                  "textAlign": "center",
                }
              }
            >
              CANCELAR
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
